# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/FileManager.h
# Commit: 74d93e32c149
# Full Hash: 74d93e32c14908ff1d29cc911eca21a82f177551
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-05-06 16:28:18
# Regressor Bug: 1623278
# File Overlap Count: 1
# Description:
#   Bug 1623278 - Use SafeRefCounted in ActorsParent. r=dom-workers-and-storage-reviewers,janv
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69351
# ==============================================================================

diff -r 6efa98635cf4 -r 74d93e32c149 dom/indexedDB/FileManager.h
--- a/dom/indexedDB/FileManager.h	Wed May 06 12:20:32 2020 +0000
+++ b/dom/indexedDB/FileManager.h	Wed May 06 11:35:20 2020 +0000
@@ -19,7 +19,8 @@
 namespace indexedDB {
 
 // Implemented in ActorsParent.cpp.
-class FileManager final : public FileManagerBase<FileManager> {
+class FileManager final : public FileManagerBase<FileManager>,
+                          public AtomicSafeRefCounted<FileManager> {
   using PersistenceType = mozilla::dom::quota::PersistenceType;
   using FileManagerBase<FileManager>::MutexType;
 
@@ -85,11 +86,10 @@
 
   [[nodiscard]] nsresult AsyncDeleteFile(int64_t aFileId);
 
-  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(FileManager)
+  MOZ_DECLARE_REFCOUNTED_TYPENAME(FileManager)
 
   static StaticMutex& Mutex() { return sMutex; }
 
- private:
   ~FileManager() = default;
 };
 